x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/api/connector/delete_connector_api_logic.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DeleteConnectorResponse } from '../../../../../common/types/connectors';
+
+import { Actions, createApiLogic } from '../../../shared/api_logic/create_api_logic';
+import { HttpLogic } from '../../../shared/http';
+
+export interface DeleteConnectorApiLogicArgs {
+  connectorId: string;
+  indexNameToDelete: string | null;
+}
+
+export interface DeleteConnectorApiLogicResponse {
+  acknowledged: boolean;
+}
+
+export const deleteConnector = async ({
+  connectorId,
+  indexNameToDelete,
+}: DeleteConnectorApiLogicArgs) => {
+  return await HttpLogic.values.http.delete(
+    `/internal/enterprise_search/connectors/${connectorId}`,
+    {
+      query: {
+        indexNameToDelete,
```

Comment:
Maybe we should figure what is the index in the endpoint rather than sending it from FE.

Reply:
Yeah that would make more sense to me